### PR TITLE
chore(server): replica not accumulate multi commands untill exec

### LINKED ISSUE #2544: replica execute multi as single
introduce a new flag, which under this mode we will execute multi commands in replica without waiting for exec to be received

### PR DESCRIPTION
**Pull request by adiholden**
fixes #2544 
when enable_multi_shard_sync flag is false we do not sync execution of multi commands between shards in replication.
Now we will not accumulate the multi commands as well but will just execute them once we get the journal change in replica

### CODE REVIEW

**Review by BorysTheDev — APPROVED** [review]

File: src/server/replica.cc [hunk 1]
```diff
@@ -803,17 +803,25 @@ void DflyShardReplica::StableSyncDflyReadFb(Context* cntx) {
...
-    if (tx_data->opcode != journal::Op::PING) {
+    if (tx_data->opcode == journal::Op::PING) {
+      force_ping_ = true;
+      journal_rec_executed_.fetch_add(1, std::memory_order_relaxed);
+    } else if (tx_data->opcode == journal::Op::EXEC) {
       if (use_multi_shard_exe_sync_) {
         InsertTxDataToShardResource(std::move(*tx_data));
       } else {
-        ExecuteTxWithNoShardSync(std::move(*tx_data), cntx);
+        // On no shard sync mode we execute multi commands once they are recived, therefor when
```
Comment (romange) [inline src/server/replica.cc:813]:
```suggestion
        // On no shard sync mode we execute multi commands once they are recieved, therefore when
```

File: src/server/journal/tx_executor.cc [hunk 2]
```diff
@@ -47,32 +49,29 @@ void MultiShardExecution::CancelAllBlockingEntities() {
...
-    case journal::Op::MULTI_COMMAND:
-      commands.push_back(std::move(entry.cmd));
-      dbid = entry.dbid;
-      return false;
+      return;
     default:
       DCHECK(false) << "Unsupported opcode";
   }
-  return false;
+  return;
```
Comment (romange) [inline src/server/journal/tx_executor.cc:74]:
```suggestion
```

File: src/server/replica.cc [hunk 3]
```diff
@@ -803,17 +803,25 @@ void DflyShardReplica::StableSyncDflyReadFb(Context* cntx) {
...
+        // revieving exec opcode we only increase the journal counting.
```
Comment (romange) [inline src/server/replica.cc:814]:
```suggestion
        // receiving exec opcode, we only increase the journal counting.
```

**Review by romange — APPROVED** [review]

**Comment by kostasrim** [comment]
@adiholden friendly ping, Shall we merge this?

**Review by BorysTheDev — APPROVED** [review]

**Pull request merged**